### PR TITLE
Add ErrTimeoutDueToLeaderFail as expected error

### PR DESCRIPTION
**Pull request by tjungblu**
This fixes ci flakes where timeouts are expected to happen in network
partitions.

issue: https://github.com/etcd-io/etcd/issues/14333

Signed-off-by: Thomas Jungblut <tjungblu@redhat.com>

### CODE REVIEW

File: tests/integration/clientv3/connectivity/network_partition_test.go [hunk 1]
```diff
@@ -23,23 +23,29 @@ import (
 	"testing"
 	"time"
 
+	"google.golang.org/grpc"
```
Comment (tjungblu) [inline tests/integration/clientv3/connectivity/network_partition_test.go:26]: I really wonder where this import inconsistency comes from, are you guys using a different gofmt? I'm on `go version go1.18.2 linux/amd64`

Comment (serathius) [reply tests/integration/clientv3/connectivity/network_partition_test.go:26]: Etcd haven't upgraded go to 1.18

**Comment by ahrtr** [comment]
This might not be the root cause of https://github.com/etcd-io/etcd/issues/14333, because `clus.Members[0]` will not receive any message from the new leader when it loses connection with other two peers. 

But I agree that we should add `ErrTimeoutDueToLeaderFail` into the expected error list. So this PR looks good to me.

**Review by ahrtr — APPROVED** [review]
LGTM

Thank you @tjungblu

**Review by serathius — APPROVED** [review]

**Pull request merged**